src/main/java/com/example/DataService/Filters.java

```diff
@@ -0,0 +1,63 @@
+package com.example.DataService;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.example.repository.CurrencyRepository;
+
+import model.Currency;
+
+@Service
+public class Filters {
+	
+	private CurrencyRepository cr;
+	
+	public Filters(CurrencyRepository cr) {
+		this.cr = cr;
+	}
+	
+	public List<Currency> filterByRange(float minRange, float maxRange) {	
+		return cr.findAll().stream()
+				.filter(c -> c.getPriceusd() > minRange)
+				.filter(c -> c.getPriceusd() < maxRange)
+				.collect(Collectors.toList());
+	}
+	
+	
+	public List<Currency> filterTop3(String filter) {
+
+		if (filter.equals("price")) {
+			return cr.findAll().stream()
+					.sorted(Comparator.comparingDouble(Currency::getPriceusd))
+					.limit(3)
+					.collect(Collectors.toList());
+		}
+
+		if (filter.equals("changepercent")) {
```

Comment:
Do not add raw string literals to the code. Either create constants or enum to represent filter values.